packages/theme/src/cli/services/push.ts

```diff
@@ -339,18 +364,40 @@ export async function createOrSelectTheme(adminSession: AdminSession, flags: Pus
       },
     })
 
-    if (await confirmPushToTheme(selectedTheme.role as Role, flags.allowLive, adminSession.storeFqdn)) {
-      return selectedTheme
-    }
+    const confirmed = await confirmPushToTheme(
+      selectedTheme.role as Role,
+      flags.allowLive,
+      session.storeFqdn,
+      environment,
+      multiEnvironment,
+    )
+    return confirmed ? selectedTheme : undefined
   }
 }
 
-async function confirmPushToTheme(themeRole: Role, allowLive: boolean | undefined, storeFqdn: string) {
+async function confirmPushToTheme(
+  themeRole: Role,
+  allowLive: boolean | undefined,
+  storeFqdn: string,
+  environment?: string[],
+  multiEnvironment?: boolean,
+) {
   if (themeRole === LIVE_THEME_ROLE) {
     if (allowLive) {
       return true
     }
 
+    if (multiEnvironment) {
```

Comment:
Since multi env commands are non-interactive, trying to push to a live theme without `allow-live` will prevent uploads and error